Lib/fontTools/merge/tables.py

```diff
@@ -278,6 +286,18 @@ def merge(self, m, tables):
                 newcs.charStrings[name] = i
                 newcs.charStringsIndex.items.append(None)
         for name in cs.charStrings:
+            if widthsDiffer:
+                c = cs[name]
+                defaultWidthXToken = object()
+                extractor = T2WidthExtractor([], [], nominalWidthX, defaultWidthXToken)
+                extractor.execute(c)
+                width = extractor.width
+                if width != defaultWidthXToken:
```

Comment:
since you want to test for identity with the unique `object()` you may as well us `is` operator: 
```suggestion
                if width is not defaultWidthXToken:
```